app/employee-app/backend/service.bal

```diff
@@ -0,0 +1,752 @@
+//
+// Copyright (c) 2005-2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+// 
+import employee_app.authorization;
+import employee_app.database;
+import employee_app.email;
+import employee_app.types;
+
+import ballerina/cache;
+import ballerina/http;
+import ballerina/log;
+import ballerina/regex;
+import ballerina/time;
+
+configurable email:EmailAlertConfig offerEmailConfig = ?;
+
+final cache:Cache userInfoCache = new (capacity = 100, evictionFactor = 0.2);
+final cache:Cache orgStructureCache = new (capacity = 100, evictionFactor = 0.2);
+
+@display {
+    label: "Employee Application",
+    id: "hris/employee-application"
+}
+service http:InterceptableService / on new http:Listener(9090) {
+
+    # Request interceptor.
+    # + return - authorization:JwtInterceptor
+    public function createInterceptors() returns [authorization:JwtInterceptor] {
+        return [new authorization:JwtInterceptor()];
+    }
+
+    # Get basic information of a given active employee.
+    #
+    # + email - Email of the employee
+    # + return - Basic information of the employee or an error
+    resource function get employees/[string email]()
+        returns types:Employee|http:BadRequest|http:InternalServerError|http:NotFound|error {
```

Comment:
Do we need to return the `error` here? I'd suggest using a [response error interceptor (`http: ResponseErrorInterceptor`)](https://ballerina.io/learn/by-example/http-error-handling/) to handle various status code responses.

Reply:
Removed error and handled the custom errors.